app/services/fetch_coronavirus_statistics_service.rb

```diff
@@ -113,12 +114,34 @@ def latest_admissions(data)
 
   def latest_tests(data)
     latest_tests = data.find { |d| d["newPositiveTests"] }
-
     return {} unless latest_tests
 
+    daily_tests = daily_tests(latest_tests)
+    weekly_tests = weekly_tests(latest_tests, data)
+
+    daily_tests.merge!(weekly_tests)
+  end
+
+  def daily_tests(latest_tests)
     {
       new_positive_tests: latest_tests["newPositiveTests"],
       new_positive_tests_date: Date.parse(latest_tests["date"]),
     }
   end
+
+  def weekly_tests(latest_tests, data)
+    weeks_with_tests = []
+    data.each do |day_cases|
```

Comment:
I feel like there should be a way to use `select` here to clean it up a little bit but I don't know of a way to do what you want without iterating twice which we obviously don't want! Just doing a first pass through at the moment to understand what's going on :)